### DISCUSSION TITLE
ISO 8601 dates in changelog

### DISCUSSION THREAD

**Discussion by ZhishCat** [FarGroup/FarManager#626] [General]
What about to convert changelog dates to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format and how time-consuming it will be? Considering that you already have 1365 builds in github repo it probably don't worth it.

Here is example of what I'd want to see:
```
--------------------------------------------------------------------------------
drkns 2023-02-12 00:38:45+00:00 - build 6102
--------------------------------------------------------------------------------
drkns 2023-02-06 22:08:52+00:00 - build 6101
--------------------------------------------------------------------------------
drkns 2023-02-05 13:01:34+00:00 - build 6100
--------------------------------------------------------------------------------
And so on... Up until:
--------------------------------------------------------------------------------
t-rex 2006-09-15 04:43:53+03:00 - build 145
```

**Comment by alabuzhev** [comment]
I guess one can regex replace `^(\w+) (\d{2})\.(\d{2})\.(\d{4}) (\d{2}:\d{2}:\d{2}) ([+-]\d{2})(\d{2})(.+)` with `$1 $4-$3-$2 $5$6$7$8` or something like that.
But why?

**Reply by ZhishCat** [reply]
Date numbers are placed in order of significance in ISO 8601 format. You already use ISO 8601 partially e.g Far Manager's [official download page](https://farmanager.com/download.php?l=en) and archive names:
```
Far30b6060.ARM64.20221208.7z
Far30b6060.x64.20221208.7z
Far30b6060.x86.20221208.7z
```
It's just a suggestion, even though it's a pure nightmare. If dev team won't like this suggestion you can close this discussion @alabuzhev.

**Comment by alabuzhev** [comment]
> it's a pure nightmare

Not at all - as I mentioned, it can be changed with a regex replace rather easily, I'm just not sure what are the benefits: I agree that ISO 8601 is superior to both m-d-y and d-m-y in general, but these dates are purely informational, we don't use them for anything else.

**Comment by MKadaner** [comment]
Just for the record, the regex should be `^(\w+) (\d{2})\.(\d{2})\.(\d{4}) (\d{2}:\d{2}:\d{2}) ([+-]\d{2})(\d{2})(.*)` to handle build-less records.

**Comment by ZhishCat** [comment]
> MKadaner: Just for the record, the regex should be `^(\w+) (\d{2})\.(\d{2})\.(\d{4}) (\d{2}:\d{2}:\d{2}) ([+-]\d{2})(\d{2})(.*)` to handle build-less records.

It didn't cover some lines, so it won't be that easy.

> alabuzhev: I'm just not sure what are the benefits: I agree that ISO 8601 is superior to both m-d-y and d-m-y in general, but these dates are purely informational, we don't use them for anything else.

I've just wanted to give food for thought to dev team, nothing more. If to follow ISO 8601 notations you'll get no benefits rather than more logical date format.

For example, both of these lines take same amount of disk space:
```
drkns 2023-02-12T00:38:45+00:00 - build 6102
drkns 12.02.2023 00:38:45 +0000 - build 6102
```

That's it. Hopefully at some day entire humanity will use ISO 8601 date format.

**Comment by alabuzhev** [comment]
Ok, if someone is willing to go through all 28 changelogs in the repo:

[enc/changelog](https://github.com/FarGroup/FarManager/blob/master/enc/changelog)
[enc/tools/changelog](https://github.com/FarGroup/FarManager/blob/master/enc/tools/changelog)
[far/changelog](https://github.com/FarGroup/FarManager/blob/master/far/changelog)
[misc/fexcept/changelog](https://github.com/FarGroup/FarManager/blob/master/misc/fexcept/changelog)
[misc/msi-installer/changelog](https://github.com/FarGroup/FarManager/blob/master/misc/msi-installer/changelog)
[plugins/align/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/align/changelog)
[plugins/arclite/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/arclite/changelog)
[plugins/arclite/example/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/arclite/example/changelog)
[plugins/autowrap/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/autowrap/changelog)
[plugins/brackets/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/brackets/changelog)
[plugins/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/changelog)
[plugins/common/CRT/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/common/CRT/changelog)
[plugins/compare/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/compare/changelog)
[plugins/drawline/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/drawline/changelog)
[plugins/editcase/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/editcase/changelog)
[plugins/emenu/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/emenu/changelog)
[plugins/farcmds/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/farcmds/changelog)
[plugins/filecase/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/filecase/changelog)
[plugins/ftp/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/ftp/changelog)
[plugins/helloworld/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/helloworld/changelog)
[plugins/hlfviewer/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/hlfviewer/changelog)
[plugins/luamacro/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/luamacro/changelog)
[plugins/macroview/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/macroview/changelog)
[plugins/multiarc/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/multiarc/changelog)
[plugins/network/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/network/changelog)
[plugins/proclist/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/proclist/changelog)
[plugins/samefolder/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/samefolder/changelog)
[plugins/tmppanel/changelog](https://github.com/FarGroup/FarManager/blob/master/plugins/tmppanel/changelog)

and replace "12.02.2023 00:38:45 +0000" with "2023-02-12 00:38:45+00:00" (but not "2023-02-12**T**00:38:45+00:00" please, that is just unreadable) everywhere, and also update [misc/changelog/ChangelogHeader.lua](https://github.com/FarGroup/FarManager/blob/master/misc/changelog/ChangelogHeader.lua) accordingly, I'll happily merge such a PR, because why not.

**Reply by MKadaner** [reply]
#629 please.

**Comment by ZhishCat** [comment]
I've noticed that skipik doesn't seem to abide new date and time format. Over time it'll cause a minor inconsistency.
```
Far30b6108.x64.20230224\changelog
╔═════════════════ Occurrences: 5, lines: 5 ══════════════════╗
║  7:1         │skipik 23.02.2023 13:50:41 +0300 - build 6107 ║ <---
║  928:1       │skipik 2022-02-23 19:46:25+03:00 - build 5962 ║
║  983:1       │skipik 2022-01-07 14:11:12+03:00 - build 5953 ║
║  993:1       │skipik 2022-01-03 22:53:17+03:00 - build 5951 ║
║  1130:1      │skipik 2021-12-01 21:42:37+03:00 - build 5925 ║
╚══════════ Ctrl+Enter F5 Gray + Ctrl+Up Ctrl+Down ═══════════╝
```

Just tell all the contributors to update their ChangelogHeader.lua and ChangelogCheckEnglish.lua macros to avoid such cases in future.

**Reply by MKadaner** [reply]
#643

**Comment by ZhishCat** [comment]
Thanks to all contributors that made this change possible! Here is nothing to discuss on that topic anymore, so discussion is closed.
